graphql_compiler/query_formatting/common.py

```diff
@@ -90,27 +94,36 @@ def _deserialize_anonymous_json_argument(expected_type: GraphQLScalarType, value
     if not isinstance(value, expected_python_types):
         correct_type = False
     if not correct_type:
-        raise ValueError(
-            "Unexpected type {}. Expected one of {}.".format(type(value), expected_python_types)
-        )
+        raise _raise_invalid_type_error(name, expected_python_types, value)
+
+    custom_deserialization_functions = merge_non_overlapping_dicts(
```

Comment:
I think that there is opportunity for clean up here, but I tried to keep the number of changes minimal to not make this PR any longer.

Reply:
This is what previously did in this function:
1. First we access a dict to check if the GraphQL type is one of the supported type and get the expected python types for the argument.
2. Then we check if the argument value has the expected python type.
3. Then we run through a series of if-statements to determine the deserialization function. (This might involve another dict access).

I think that it makes more sense to just create a dict that maps the supported GraphQL types to their expected python types and their deserialization function.